tests/unit/test_charm.py

```diff
@@ -113,60 +101,56 @@ def test_on_leader_elected(self):
     @patch("charm.PostgresqlOperatorCharm._get_postgres_password")
-    def test_on_start(self, _get_postgres_password, _bootstrap_cluster):
-        # Test before the superuser password is generated.
+    def test_on_start(self, _get_postgres_password, _replication_password, _bootstrap_cluster):
+        # Test before the passwords are generated.
         _get_postgres_password.return_value = None
         self.charm.on.start.emit()
         _bootstrap_cluster.assert_not_called()
-        self.assertEqual(
-            self.harness.model.unit.status, WaitingStatus("waiting superuser password generation")
-        )
+        self.assertTrue(isinstance(self.harness.model.unit.status, WaitingStatus))
 
-        # Mock the superuser password.
-        _get_postgres_password.return_value = "random-password"
-
-        # Test the possible errors.
-        errors = [
-            {
-                "error": ClusterAlreadyRunningError,
-                "message": "there is already a running cluster",
-            },
-            {
-                "error": ClusterCreateError("test"),
-                "message": "failed to create cluster with error test",
-            },
-            {
-                "error": ClusterNotRunningError("test"),
-                "message": "failed to start cluster with error test",
-            },
-            {
-                "error": ClusterStartError("test"),
-                "message": "failed to start cluster with error test",
-            },
-        ]
-        for error in errors:
-            _bootstrap_cluster.side_effect = error["error"]
-            self.charm.on.start.emit()
-            _bootstrap_cluster.assert_called_once()
-            # Check the correct error message on unit status.
-            self.assertEqual(
-                self.harness.model.unit.status,
-                BlockedStatus(error["message"]),
-            )
-            # Reset the mock call count.
-            _bootstrap_cluster.reset_mock()
+        # Mock the passwords.
+        _get_postgres_password.return_value = "fake-postgres-password"
+        _replication_password.return_value = "fake-replication-password"
 
-        # Then test the event of a correct cluster bootstrapping.
-        _bootstrap_cluster.side_effect = None
+        # Mock cluster start success values.
+        _bootstrap_cluster.side_effect = [False, True]
+
+        # Test for a failed cluster bootstrapping.
         self.charm.on.start.emit()
         _bootstrap_cluster.assert_called_once()
+        self.assertTrue(isinstance(self.harness.model.unit.status, BlockedStatus))
+
+        # Set an initial waiting status (like after the install hook was triggered).
+        self.harness.model.unit.status = WaitingStatus("fake message")
+
+        # Then test the event of a correct cluster bootstrapping.
+        self.charm.on.start.emit()
         self.assertEqual(
             self.harness.model.unit.status,
             ActiveStatus(),
         )
```

Comment:
I like that you're asserting the unit status is an instance of BlockedStatus (or whatever status it is) in the rest of the tests, since it makes the test more robust to changes in status messages. Could you do the same here?

```suggestion
self.assertTrue(isinstance(self.harness.model.unit.status, ActiveStatus))
```

Reply:
Updated on https://github.com/canonical/postgresql-operator/commit/d2d0915c789894145f29c296de6b4ec5492cf382.